wgpu-core/src/resource.rs

```diff
@@ -785,6 +836,23 @@ crate::impl_parent_device!(Buffer);
 crate::impl_storage_item!(Buffer);
 crate::impl_trackable!(Buffer);
 
+impl WebGpuError for CreateBufferError {
+    fn webgpu_error_type(&self) -> ErrorType {
+        let e: &dyn WebGpuError = match self {
+            Self::Device(e) => e,
+            Self::AccessError(e) => e,
+            Self::IndirectValidationBindGroup(e) => e,
```

Comment:
issue: This should actually be `Internal`; the inner error type here doesn't have the right context to distinguish.

Reply:
Fixed with 2c81fb89.